cpp/src/structs/structs_column_factories.cu

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <algorithm>
+
+namespace cudf
+{
+  namespace 
+  {
+    // Helper function to superimpose validity of parent struct
+    // over all member fields (i.e. child columns).
+    void superimpose_validity(
+      rmm::device_buffer const& parent_null_mask, 
+      size_type parent_null_count,
+      std::vector<std::unique_ptr<column>>& children,
+      cudaStream_t stream,
+      rmm::mr::device_memory_resource* mr
+    )
+    {
+      if (parent_null_mask.is_empty()) {
+        // Struct is not nullable. Children do not need adjustment.
+        // Bail.
+        return;
+      }
+
+      std::for_each(
+        children.begin(),
+        children.end(),
+        [&](std::unique_ptr<column>& p_child)
+        {
+          if (!p_child->nullable())
+          {
+            p_child->set_null_mask(std::move(rmm::device_buffer{parent_null_mask, stream, mr})); 
+            p_child->set_null_count(parent_null_count);
+          }
+          else {
+
+            auto data_type{p_child->type()};
+            auto num_rows{p_child->size()};
+
+            // All this to reset the null mask. :/
+            cudf::column::contents contents{p_child->release()};
```

Comment:
You don't need to `release` the contents of the child just to reset it's null mask. Just use [`set_null_mask`](https://docs.rapids.ai/api/libcudf/nightly/classcudf_1_1column.html#ab057d715c84c850103b5b047091bcb0a).

Reply:
Oh, shoot. Thanks for the heads-up, @jrhemstad. I'm not sure how I missed it when I looked for the setter.

Reply:
Thanks for the pointer, @devavret. (Please excuse the pun.) I'm now fetching the null_mask from the column's `mutable_view()`.